SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -459,6 +463,16 @@ void CIncNSSolver::BC_Wall_Generic(const CGeometry *geometry, const CConfig *con
         LinSysRes(iPoint, nDim+1) += scalar_factor*dot_product;
       } // if streamwise_periodic
     }
+    else if (kind_boundary == HEAT_TRANSFER) {
+      Twall = nodes->GetTemperature(iPoint);
+      Wall_HeatFlux = Transfer_Coefficient * (Tinfinity - Twall);
+
+      /*--- Apply a weak boundary condition for the energy equation.
+      Compute the residual due to the prescribed temperature and transfer coefficient.
+      Note that for the Heat_Transfer wall, basically a heatflux wall that depends on the local Temperature is applied. ---*/
+      LinSysRes(iPoint, nDim+1) -= Wall_HeatFlux*Area;
```

Comment:
You may want to update the Jacobian accordingly, it may be why you get convergence problems with high "h"

Reply:
Yes good point, I thought about that at some moment but forgot about it 👍

Reply:
Good catch again, this indeed resolves the convergence issues with high values of the transfer coefficient 👍 I was now able to easily run the case (shown in a picture above) with h=1e8 instead of a max of h=1e5. That now results in a surface avg surface static temperature at exactly 400K which matches the expectation. Awesome